### ISSUE TITLE
Allow to make rest parameters to force non-zero length

### ISSUE THREAD

**Issue by rinart73** [sveltejs/kit#10305]
### Describe the problem

I'm trying to make a multilingual website with localized URLs.
I have been advised to use the following structure:

* `[...news=news]`
  * `[slug]`

**src/params/news.ts:**
```typescript
import type { ParamMatcher } from '@sveltejs/kit';

export const match: ParamMatcher = (param) => {
    return param === 'news' || param === 'ru/novosti';
};
```
Expected behaviour:

* `/news` - News list page in English
* `/news/foo` - "Foo" news post in English from DB
* `/ru/novosti` - News list page in Russian
* `/ru/novosti/foo` - "Foo" news post in Russian from DB
* `/ru/news` - 404
* `/novosti` - 404
* `/barbaz` - 404

Actual behaviour:

* `/novosti` - "Novosti" news post in English
* `/barbaz` - "Barbaz" news post in English

If I understand correctly this happens because in the past rest parameters were changed to allow zero-length ( https://github.com/sveltejs/kit/issues/554 ), in which case a corresponding matcher doesn't run on them.

### Describe the proposed solution

Maybe allow to specify which rest parameters can have zero-length and which can't?

### Alternatives considered

Initially I thought of a simpler solution, but it's currently not supported because there is no way to read already matched "parent" paramerers ( https://github.com/sveltejs/kit/issues/4500 ).
* `[lang=lang]`
  * `[news=news]`
    * `[slug]`

If everything else fails I guess I'll have to abandon the idea of localizing URLs, even though it's nice for usability.

### Importance

would make my life easier

**Comment by Conduitry** [comment]
Does checking the rest param in your `load` doing a `throw error(404, 'whatever')` if it's zero-length work?

**Comment by rinart73** [comment]
Not really, because it would use `+layout.svelte` and `+error.svelte` from the `routes/[...news=news]/[slug]/` directory instead of the expected `routes/`.
